infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironment.java

```diff
@@ -356,6 +403,14 @@ public void setMetadata(ObjectMeta podMeta) {
       this.podMeta = podMeta;
     }
 
+    public PodRole getRole() {
+      return role;
+    }
+
+    public void setRole(PodRole role) {
+      this.role = role;
+    }
+
```

Comment:
may be helper methods like `isInjectable()` and `isDeployment()` cus i see a lot of `getRole() ==...` constructions

Reply:
Not sure that it would improve the readability if we had `pod.isDeployment()` instead of `pod.getRole() == PodRole.DEPLOYMENT`. IMHO the latter is actually more clear in the intention.